internal/controllers/packages/package_controller_test.go

```diff
@@ -0,0 +1,262 @@
+package packages
```

Comment:
The unit tests here look really good, thanks! But there is no test for the pause feature. I think that would be nice to have.